connectors/src/lib/models/webcrawler.ts

```diff
@@ -0,0 +1,174 @@
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    url: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+  },
+  {
+    sequelize: sequelize_conn,
+    indexes: [],
+    modelName: "webcrawler_configurations",
+  }
+);
+Connector.hasMany(WebCrawlerConfiguration);
+
+export class WebCrawlerFolder extends Model<
+  InferAttributes<WebCrawlerFolder>,
+  InferCreationAttributes<WebCrawlerFolder>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+  declare parentUrl: string | null;
+  declare url: string;
+  // Folders are not upserted to the data source but their ids are
+  // used as parent to WebCrawlerPage.
+  declare dustDocumentId: string;
+  declare connectorId: ForeignKey<Connector["id"]>;
+  declare webcrawlerConfigurationId: ForeignKey<WebCrawlerConfiguration["id"]>;
+}
+
+WebCrawlerFolder.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    url: {
+      type: DataTypes.TEXT,
+      allowNull: false,
+    },
+    parentUrl: {
+      type: DataTypes.TEXT,
+      allowNull: true,
+    },
+    dustDocumentId: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+  },
+  {
+    sequelize: sequelize_conn,
+    indexes: [
+      {
+        unique: true,
+        fields: ["url", "connectorId", "webcrawlerConfigurationId"],
+      },
+    ],
+    modelName: "webcrawler_folders",
+  }
+);
+Connector.hasMany(WebCrawlerFolder);
+WebCrawlerConfiguration.hasMany(WebCrawlerFolder);
+
+export class WebCrawlerPage extends Model<
+  InferAttributes<WebCrawlerPage>,
+  InferCreationAttributes<WebCrawlerPage>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+  declare title: string | null;
+  declare parentUrl: string | null;
+  declare url: string;
+  declare dustDocumentId: string;
+  declare connectorId: ForeignKey<Connector["id"]>;
+  declare webcrawlerConfigurationId: ForeignKey<WebCrawlerConfiguration["id"]>;
+}
+
+WebCrawlerPage.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    url: {
+      type: DataTypes.TEXT,
+      allowNull: false,
+    },
+    title: {
+      type: DataTypes.TEXT,
+      allowNull: true,
+    },
+    parentUrl: {
+      type: DataTypes.TEXT,
+      allowNull: true,
+    },
+    dustDocumentId: {
```

Comment:
Let's call it internalId and or for Pages documentId ? the dust prefix is not super useful here?

Reply:
I like having the Dust prefix. I see connectors as a client of the Dust API, hence storing the dust id.
If we call it `internalId` it creates ambiguity for it to either be an connectors level id or a dust id.

Reply:
Let's just call it `documentId` ?